File: Python/Leetcode/Find the Highest Altitude/find_the_highest_altitude.py
class Solution:
    def largestAltitude(self, gain) -> int:
        ans,sum_all = 0,0
        for i in gain:
            sum_all+=i
            ans=max(sum_all,ans)
        return ans

item  = Solution()
print(item.largestAltitude([-5,1,5,0,-7]))

File: Leetcode/1732. Find the Highest Altitude/find_the_highest_altitude.rs
struct Solution {}

impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut largest = 0;
        let mut sum_all = 0;
        for current_gain in gain{
            sum_all+=current_gain;
            if sum_all>largest{
                largest = sum_all;
            }
        }
        largest
    }
}

fn main() {
    let res = Solution::largest_altitude(vec![-4,-3,-2,-1,4,3,2]);
    println!("{:?}", res);
}
